File: RedSheet.DbEntities/Models/AvoidSayingOpinion.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("AvoidSayingOpinions", Schema = "dbo")]
	[Serializable]
	public partial class AvoidSayingOpinion  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int AvoidSayingOpinionId { get; set; } 
 
	[Required] 
	[MaxLength(500)]
	public string AvoidSayingOpinionValue { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int CultureCountryId { get; set; } 

	public AvoidSayingOpinion()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/avoid-saying-opinion.ts
import { required, maxLength, range } from '@rx/annotations';

export class AvoidSayingOpinion {
    constructor(avoidSayingOpinion?: AvoidSayingOpinion )  {
        let properties = [ "avoidSayingOpinionId", "avoidSayingOpinionValue", "cultureCountryId",];
        for (let property of properties)
            if (avoidSayingOpinion && avoidSayingOpinion[property])
                this[property] = avoidSayingOpinion[property];
    }
 
	avoidSayingOpinionId : number =   0 ;
 
    @required()
    @maxLength(500)
	avoidSayingOpinionValue : string =   undefined;
 
    @range(1,2147483647)
	cultureCountryId : number =   undefined;
}
